src/test/java/pl/javastart/streamstask/StreamsTaskTest.java

```diff
@@ -0,0 +1,164 @@
+    private Expense expense5;
+
+    @BeforeEach
+    void init() {
+        streamsTask = new StreamsTask();
+
+        users = new ArrayList<>();
+
+        user1 = new User(1L, "Alicja", 20);
+        user2 = new User(2L, "Dominik", 15);
+
+        users.add(user1);
+        users.add(user2);
+        users.add(new User(3L, "Patrycja", 25));
+        users.add(new User(4L, "Marcin", 30));
+        users.add(new User(5L, "Tomek", 18));
+        users.add(new User(6L, "Damian", 26));
+
+        expenses = new ArrayList<>();
+
+        expense1 = new Expense(1L, "Buty", new BigDecimal("149.99"), ExpenseType.WEAR);
+        expense2 = new Expense(1L, "Sałatka", new BigDecimal("14.99"), ExpenseType.FOOD);
+        expense3 = new Expense(2L, "Bluza", new BigDecimal("100"), ExpenseType.WEAR);
+        expense4 = new Expense(2L, "Skarpetki", new BigDecimal("39"), ExpenseType.WEAR);
+        expense5 = new Expense(2L, "Pizza", new BigDecimal("25"), ExpenseType.FOOD);
+
+        expenses.add(expense1);
+        expenses.add(expense2);
+        expenses.add(expense3);
+        expenses.add(expense4);
+        expenses.add(expense5);
+    }
+
+    @Test
+    void shouldFilterWomenCorrectly() {
+
+        // when
+        Collection<User> women = streamsTask.findWomen(users);
+
+        // then
+        for (User woman : women) {
+            assertThat(woman.getName()).endsWith("a");
+        }
+    }
+
+    @Test
+    void shouldCalculateAverageAgeForMen() {
+
+        // when
+        Double avgManAge = streamsTask.averageMenAge(users);
+
+        // then
+        assertThat(avgManAge).isEqualTo(22.25);
+    }
+
+    @Test
+    void shouldGroupExpensesByUserId_v1() {
+
+        // when
+        Map<Long, List<Expense>> expensesByUserId = streamsTask.groupExpensesByUserIdVersion1(expenses);
+        String user1ExpensesConcatenatedNames = getConcatenatedExpenseNames(1L, expensesByUserId);
+        String user2ExpensesConcatenatedNames = getConcatenatedExpenseNames(2L, expensesByUserId);
+
+        // then
+        assertThat(expensesByUserId.size()).isEqualTo(2);
+        assertThat(expensesByUserId.get(1L).size()).isEqualTo(2);
+        assertThat(expensesByUserId.get(2L).size()).isEqualTo(3);
+        assertThat(user1ExpensesConcatenatedNames).contains("Buty");
+        assertThat(user1ExpensesConcatenatedNames).contains("Sałatka");
+        assertThat(user2ExpensesConcatenatedNames).contains("Bluza");
+        assertThat(user2ExpensesConcatenatedNames).contains("Skarpetki");
+        assertThat(user2ExpensesConcatenatedNames).contains("Pizza");
+    }
+
+    @Test
+    void shouldGroupExpensesByUserId_v2() {
+
+        // when
+        Map<Long, List<Expense>> expensesByUserId = streamsTask.groupExpensesByUserIdVersion2(users, expenses);
+        String user1ExpensesConcatenatedNames = getConcatenatedExpenseNames(1L, expensesByUserId);
+        String user2ExpensesConcatenatedNames = getConcatenatedExpenseNames(2L, expensesByUserId);
```

Comment:
Nie do końca widzę sens tworzenia tego Stringa. Lepiej byłoby się posługiwać po prostu kolekcjami, choćby tak, jak zrobiłeś to w kolejnych testach